connect/runtime/src/main/java/org/apache/kafka/connect/runtime/errors/ErrorReporter.java

```diff
@@ -16,17 +16,25 @@
  */
 package org.apache.kafka.connect.runtime.errors;
 
+import org.apache.kafka.clients.producer.RecordMetadata;
+
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Future;
+
 /**
  * Report an error using the information contained in the {@link ProcessingContext}.
  */
 public interface ErrorReporter extends AutoCloseable {
 
     /**
-     * Report an error.
+     * Report an error and return the producer future.
      *
      * @param context the processing context (cannot be null).
+     * @return future result from the producer sending a record to Kafka.
      */
-    void report(ProcessingContext context);
+    default Future<RecordMetadata> report(ProcessingContext context) {
+        return CompletableFuture.completedFuture(null);
+    }
```

Comment:
We should not make this method a default method, since both implementations of the interface define this method.